### PR TITLE
Store entity manager controller as a class member and use it to load/render entities

### PR DESCRIPTION
**Pull request by cs-shadow**
Details in this issue: https://drupal.org/node/2271203

### CODE REVIEW

File: lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php [hunk 1]
```diff
@@ -42,11 +69,18 @@ public function process($text, $langcode, $cache, $cache_id) {
           // Load the entity either by UUID (preferred) or ID.
           if ($node->hasAttribute('data-entity-uuid')) {
             $uuid = $node->getAttribute('data-entity-uuid');
-            $entity = entity_load_by_uuid($entity_type, $uuid);
+
+            $entity_type_definition = \Drupal::entityManager()->getDefinition($entity_type);
```
Comment (davereid) [inline lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php]: Shouldn't this use $this->entity_manager?

Comment (cs-shadow) [inline lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php]: It should be. My bad. Fixed in latest commit.

**Comment by slashrsm** [comment]
Looks good but apparently there are merge conflicts. Please re-roll and it should be ready to go in.

**Comment by cs-shadow** [comment]
Re-rolled against latest HEAD.

File: lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php [hunk 2]
```diff
@@ -137,7 +172,9 @@ public static function postRender(array $element, array $context) {
         }
 
         // Build the rendered entity.
-        $build = entity_view($entity, $context['view_mode'], $context['langcode']);
+        $entity_type_id = $entity->getEntityTypeId();
+        $render_controller = $this->entity_manager->getViewBuilder($entity_type_id);
```
Comment (davereid) [inline lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php:176]: This was not actually tested and does not work. This is a public static function and does not have a $this-> available.

Comment (davereid) [inline lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php:176]: I had to revert the changes in postRender() for now otherwise the module wasn't even working.

Comment (cs-shadow) [inline lib/Drupal/entity_embed/Plugin/Filter/EntityEmbedFilter.php:176]: Sorry for the trouble. I forgot to clear the cache while testing and it worked. After clearing all caches it gives on error.
